core/src/main/scala/latis/ops/TimeTupleToTime.scala

```
@@ -0,0 +1,79 @@
+package latis.ops
+
+import cats.implicits._
+import latis.data._
+import latis.metadata.Metadata
+import latis.model._
+import latis.time.Time
+import latis.util.LatisException
+
+/**
+ * Defines an Operation that converts a Tuple of time values to a single time Scalar.
+ *
+ * @param name the name of the Tuple that stores the time values
+ */
+case class TimeTupleToTime(name: String = "time") extends MapOperation {
+
+  override def applyToModel(model: DataType): DataType = {
+    val timeTuple = model.findAllVariables(name) match {
+      case Nil               => throw new LatisException(s"Cannot find variable: $name")
+      case vs: Seq[DataType] => vs.head
+    }
+
+    val time: Scalar = timeTuple match {
+      case Tuple(es @ _*) =>
+        //build up format string
+        val format: String = es.toList.traverse(_("units"))
+          .fold(throw new LatisException("A time Tuple must have units defined for each element."))(_.mkString(" "))
+        //make the time Scalar
+        val metadata = Metadata("id" -> "time", "units" -> format, "type" -> "string")
+        Time(metadata)
+    }
+    
+    model.map {
+      case t: Tuple if t.id == name => time //TODO: support aliases with hasName?
+      case v => v
+    }
+  }
+
+  override def mapFunction(model: DataType): Sample => Sample = {
+    val timePos: SamplePosition = model.getPath(name) match {
+      case Some(List(sp)) => sp
+      case _ => throw new LatisException(s"Cannot find path to variable: $name")
```

Can `getPath` return a list of more than one element? Maybe that would require matching on aliases. If so, we might not want to throw an exception here, or at least change the message.

---

I think it technically can; the scaladoc for `getPath` says:

> Return the path within this DataType to a given variable ID
> as a sequence of SamplePositions.
> Note that length of the path reflects the number of nested Functions.

So if the time Tuple was inside a nested Function or two, we'd get our error `"Cannot find path to variable: $name"`. I'm not exactly sure how I'd have to alter the pattern match on `timePos` to handle this nested Function case, but I've also never seen a dataset that put a time Tuple in a nested Function, so I'm tempted to say that our error message here is correct... we cannot find a path to a variable in a nested function. 

Maybe it'd be appropriate to add a case between these two cases that catches a List of `SamplePositions` longer than one, then throw an error saying we don't handle time Tuples in nested Functions.

---

> Maybe it'd be appropriate to add a case between these two cases that catches a List of SamplePositions longer than one, then throw an error saying we don't handle time Tuples in nested Functions.

```
case None => throw new LatisException(s"Cannot find path to variable: $name")
case _ => throw new LatisException(s"A time Tuple in a nested Function?!")
```